Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py

```diff
@@ -2864,171 +3212,453 @@ def extract_sentinelone_zip_file(zip_file_data: bytes, password: str) -> Tuple[s
+
+
+def close_remote_incident(client: Client, threat_id: str, delta: dict) -> str:
+    """
+    This method will close the remote incident
+    """
+    verdict = ANALYST_VERDICT.get(delta.get("sentinelonethreatanalystverdict", ""), None)
+    client.update_threat_analyst_verdict_request(threat_ids=argToList(threat_id), action=verdict)
+    response = client.update_threat_status_request(threat_ids=argToList(threat_id), status="resolved")
+    if response.get("affected") and int(response.get("affected")) > 0:
+        demisto.debug("Successfully updated the threat status and marked as resolved")
+        note = "XSOAR - Marked as resolved \n" + delta.get("closeNotes", "")
+        client.write_threat_note_request(threat_ids=argToList(threat_id), note=note)
+    else:
+        demisto.debug("Unable to mark as resolved")
+    return ""
+
+
+def update_remote_system_command(client: Client, args: dict) -> str:
+    """update-remote-system command: pushes local changes to the remote system
+
+    :type client: ``Client``
+    :param client: XSOAR client to use
+
+    :type args: ``Dict[str, Any]``
+    :param args:
+        all command arguments, usually passed from ``demisto.args()``.
+        ``args['data']`` the data to send to the remote system
+        ``args['entries']`` the entries to send to the remote system
+        ``args['incidentChanged']`` boolean telling us if the local incident indeed changed or not
+        ``args['remoteId']`` the remote incident id
+        args: A dictionary containing the data regarding a modified incident, including: data, entries, incident_changed,
+         remote_incident_id, inc_status, delta
+
+    :return:
+        ``str`` containing the remote incident id - really important if the incident is newly created remotely
+
+    :rtype: ``str``
+    """
+    parsed_args = UpdateRemoteSystemArgs(args)
+    delta = parsed_args.delta
+    remote_incident_id = parsed_args.remote_incident_id
+    demisto.debug(f'Got the following data {parsed_args.data}, and delta {delta}.')
+    if delta:
+        demisto.debug(f'Got the following delta keys {list(delta.keys())}.')
+    demisto.debug(
+        f"Sending incident with remote ID [{parsed_args.remote_incident_id}] to remote system\n"
+    )
+    try:
+        if parsed_args.incident_changed:
+            if delta.get("closeNotes", None):
+                close_remote_incident(client, remote_incident_id, delta)
+            else:
+                update_remote_incident(client, remote_incident_id, delta)
+    except Exception as e:
+        demisto.error(f'Error in SentinelOne outgoing mirror for incident {remote_incident_id}. '
+                      f'Error message: {str(e)}')
+
+    return remote_incident_id
+
+
+def set_xsoar_incident_entries(mirrored_object: dict, entries: list, remote_incident_id: str):
+    demisto.debug("with in the set xsoar incident entries method")
+    if mirrored_object.get("threatInfo", {}).get("incidentStatus") == "resolved":
```

Comment:
Add an additional argument "Close Mirrored XSOAR Incident" that only if it is checked the incident will be closed.